### DISCUSSION TITLE
GSSOC'23 How to contribute to ProjectsHut? 🧡🚀

### DISCUSSION THREAD

**Discussion by priyankarpal** [priyankarpal/projectshut#415] [General]
![image](https://github.com/priyankarpal/ProjectsHut/assets/88102392/0c5debf5-d414-4916-87d8-e1a710773ae3)

Welcome to ProjectsHut, an exciting open-source project that has been selected for GirlScript Summer of Code (GSSOC) '23! We are thrilled to have you on board and are looking forward to your valuable contributions. This guide will provide you with comprehensive guidelines on how to contribute effectively to ProjectsHut during this event.

## Getting Started

> Before going to contribute please check this guide

**Understand the Project:** 
- Familiarize yourself with the project's objectives, architecture, and existing codebase. This will enable you to make informed contributions that align with the project's goals.

**Choose an Issue/Create an issue:** 
- Create an issue or work on the existing issue if you're assigned. Also, if you're a contributor of  **GSSOC'23** please add `"I'm in GSSOC23"` so that we can add the `gssoc23` label on that particular issue/PR. 

**Work on Your assigned issue first:** 
- We kindly request that you prioritize working on your assigned issue at this time. It is highly recommended that you begin by creating an issue before proceeding to create a pull request. By doing so, you can outline the problem or task at hand and facilitate a smoother workflow. Once the issue is established, please proceed to implement the necessary changes, address any bugs, or incorporate new features to effectively resolve the selected issue/task.

**Commit Your Changes:**
- Once you have made significant progress or completed your contribution, commit your changes with a descriptive commit message. It is advisable to make frequent, small commits to make the review process more manageable. Please check [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0) 

**Push Your Changes:**
- Push your local branch to your GitHub repository using the git push command. This will make your changes accessible for review and integration into the main project.

**Create a Pull Request:**
- Navigate to the original ProjectsHut repository on GitHub and create a pull request from your forked repository. Provide a clear and concise description of your changes, referencing the issue/task you addressed. Mention any relevant tests or documentation that you have added or updated.

## Review and Collaboration
**Engage in Discussions:**
- Engage in active discussions on the pull request, addressing any questions, suggestions, or feedback from the project maintainers or the community. Collaborate and iterate on your contribution based on the feedback received.

**Make Necessary Changes:** 
- Make the requested changes and push them to your branch. The pull request will update automatically, reflecting the new changes.

**Resolve Conflicts:**
- In case there are conflicts with the main project's branch, resolve them by merging or rebasing your branch onto the latest changes from the main project.

**Tests and Documentation:** 
- Ensure that your contribution includes appropriate tests to validate the functionality and any relevant updates to the project's documentation.

**Get Your Contribution Merged:**
- Once your contribution has been reviewed, approved, and meets the project's requirements, a project maintainer will merge your changes into the main codebase. Congratulations on successfully contributing to ProjectsHut!

We already have [Contributing guide](https://github.com/priyankarpal/ProjectsHut/blob/main/contributing.md) you can check it. Also, don't forget to check the [README](https://github.com/priyankarpal/ProjectsHut/blob/main/README.md)

# Final Notes
By following these contribution guidelines, you can make a meaningful impact on ProjectsHut during GSSOC'23. Remember to communicate effectively, seek help when needed, and if you've any doubts you can ask here or you can reach out to me on [Twitter]( https://twitter.com/Priyankarpal)

![image](https://github.com/priyankarpal/ProjectsHut/assets/88102392/be0914e2-acfa-4be9-84b2-b67ec6dfcd0e)

**Comment by govardhan-26** [comment]
I am a GSSOC contributor...but while creating an issue...forgot to mention it...What should I do ?

**Reply by priyankarpal** [reply]
share your issue number I will add

**Comment by Ritika8081** [comment]
I wanna work on #1035 issue but not able to comment on that

**Comment by imohit159** [comment]
i am having problem while creating new issue. please check this out
![Screenshot 2023-06-05 191000](https://github.com/priyankarpal/ProjectsHut/assets/125040386/4f8f5926-79ea-47ca-94f7-53c1ec8085d3)
![Screenshot 2023-06-05 191016](https://github.com/priyankarpal/ProjectsHut/assets/125040386/c0d96051-6e03-40c7-a824-aa9b9785892c)

**Comment by imohit159** [comment]
issue no. :- #1114 
[issue link] (https://github.com/priyankarpal/ProjectsHut/issues/1114#issue-1741791847)

**Comment by murtazajoo** [comment]
@suzy-g38 you can added min-width in the profile page 

![image](https://github.com/priyankarpal/ProjectsHut/assets/98375522/62080929-70e2-4cb4-83fc-6bf5bd927f8d)


to

![image](https://github.com/priyankarpal/ProjectsHut/assets/98375522/0511a0f7-f1ac-49df-ac08-fb755f757a58)

**Comment by Shahil093** [comment]
#1307 
Please assign me this issue
